### PR TITLE
[1.10.2] Rework some blocks, delete METADATA property

### PR DESCRIPTION
**Pull request by Exidex**
Rewrite some blocks using properties and some minor tweaks

### CODE REVIEW

File: src/main/java/gregtech/api/enums/Materials.java [hunk 1]
```diff
@@ -1570,25 +1571,6 @@ public Materials(int aMetaItemSubID, MaterialSet aIconSet, float aToolSpeed, int
...
-	 * This is for keeping compatibility with addons mods (Such as TinkersGregworks etc) that looped over the old materials enum
-	 */
-	@Deprecated
-	public static Materials valueOf(String aMaterialName) {
-		return getMaterialsMap().get(aMaterialName);
-	}
-
-	/**
-	 * This is for keeping compatibility with addons mods (Such as TinkersGregworks etc) that looped over the old materials enum
-	 */
```
Comment (JohannesGaessler) [inline src/main/java/gregtech/api/enums/Materials.java]: Are you sure we should just remove these methods? I don't know whether TinkersGregwork is still relevant but is there an actual benefit to removing the deprecated methods?

**Review by JohannesGaessler — COMMENTED** [review]
I have trouble understanding the nature of your changes.
Please describe what you changed and why.

**Comment by Exidex** [comment]
So instead of having one Property METADATA which is just a number, for example ores now have 2 properties which represent StoneType and Material and you can now access those properties of an ore without any manipulations on a some number from 0 to 15.

**Comment by draknyte1** [comment]
I assume this is for 1.10.
Changing things like this for 1.7.10 isn't really a valid idea.

**Comment by Exidex** [comment]
It is targeted at 1.10.2 branch.

File: src/main/java/gregtech/api/enums/Materials.java (hunk 1 above)
Comment (Exidex) [reply src/main/java/gregtech/api/enums/Materials.java]: There is no addons for 1.10.2 for GT and these methods have their not deprecated analogs

Comment (draknyte1) [reply src/main/java/gregtech/api/enums/Materials.java]: I'll probably use them initially when I port to 1.10.2

Comment (Exidex) [reply src/main/java/gregtech/api/enums/Materials.java]: So should I revert it?

Comment (draknyte1) [reply src/main/java/gregtech/api/enums/Materials.java]: Leaving them does no damage. I would appreciate it.)

File: src/main/java/gregtech/common/blocks/GT_Block_GeneratedOres.java [hunk 2]
```diff
@@ -34,33 +35,35 @@
...
-    public static void doOreThings() {
-        System.out.println("MATERIALS META OFFSET: " + MATERIALS_META_OFFSET);
-        System.out.println("MATERIALS/BLOCK: " + MATERIALS_PER_BLOCK);
-        sGeneratedBlocks = new IBlockState[GregTech_API.sGeneratedMaterials.length][];
-        sGeneratedSmallBlocks = new IBlockState[GregTech_API.sGeneratedMaterials.length][];
         Materials[] lastMats = new Materials[MATERIALS_PER_BLOCK];
         int length = 0;
-        for(Materials aMaterial : Materials.values()) {
-            if(aMaterial != null && aMaterial.mMetaItemSubID >= 0 && (aMaterial.mTypes & 0x08) != 0) {
+        for(Materials aMaterial : GregTech_API.sGeneratedMaterials) {
```
Comment (Exidex) [inline src/main/java/gregtech/common/blocks/GT_Block_GeneratedOres.java]: This will fix the issue that was discussed in #994

**Comment by KartAlexPet** [comment]
When will there be a new version with a solution?
The problem with rftools is still relevant.

**Comment by Exidex** [comment]
@KartAlexPet GT5U for 1.10 is now under a major rewrite, some things are done, more to do. I don't think it will be ready at least for a couple months.

**Comment by KartAlexPet** [comment]
I understood. Good luck.

**Pull request merged**